### PR TITLE
Rewrite JavaProcessStackTracesMonitor in Java

### PR DESCRIPTION
**Pull request by blindpirate**
Previously we ran into some issues when invoking JavaProcessStackTracesMonitor
from LifecyclePlugin "can't resolve dependencies in non-Gradle-managed thread".
Now we rewrite it in Java and remove all dependencies so it can be invoked directly.

### CODE REVIEW

**Comment by blindpirate** [comment]
@bot-gradle test QF

File: buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/lifecycle/LifecyclePlugin.kt [hunk 1]
```diff
@@ -134,9 +134,9 @@ class LifecyclePlugin : Plugin<Project> {
         if (BuildEnvironment.isCiServer) {
             val timer = Timer(true).apply {
                 schedule(timerTask {
-                    javaexec {
-                        classpath = project(":internalIntegTesting").configurations["runtimeClasspath"]
-                        main = "org.gradle.integtests.fixtures.timeout.JavaProcessStackTracesMonitor"
+                    exec {
+                        commandLine("${System.getProperty("java.home")}/bin/java",
+                            rootProject.file("subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/timeout/JavaProcessStackTracesMonitor.java"))
```
Comment (jjohannes) [inline buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/lifecycle/LifecyclePlugin.kt:139]: Preferably use `layout.projectDirectory.file(..)` in the future

**Comment by blindpirate** [comment]
@bot-gradle test QF

**Comment by blindpirate** [comment]
@bot-gradle test QF

**Comment by blindpirate** [comment]
@bot-gradle test QF

**Pull request merged**